global-news-app/src/components/Search/Search/Search.tsx

```diff
@@ -0,0 +1,81 @@
+import React from "react";
+import { SearchContainer } from "./Search.styles";
+import SearchInput from "../SearchInput/SearchInput";
+import Select from "components/Select/Select";
+
+import RecentSearchesDropdown from "../RecentSearchesDropdown/RecentSearchesDropdown";
+import { ClickAwayListener } from "@mui/base/ClickAwayListener";
+import { SearchSelectOptions } from "Utils/MockData";
+
+const Search = () => {
+  const [option, setOption] = React.useState("");
+  const [searchHistory, setSearchHistory] = React.useState<string[]>([]);
+  const [showDropdown, setShowDropdown] = React.useState(false);
+
+  const handleSearch = (query: string) => {
+    // Simulate calling an external API
+    if (query) {
+      simulateExternalAPICall(query);
+      // Update search history
+      setSearchHistory([...searchHistory, query]);
+      setShowDropdown(true);
+    }
+  };
+
+  const simulateExternalAPICall = (query: string) => {
+    // Simulate calling an external API
+    console.log(`Calling external API with query: ${query}`);
+  };
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === "Enter") {
```

Comment:
check if there is a type for keyboard events in order to use instead of string like "Enter"